layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -1350,10 +1350,43 @@ the right."
   (interactive)
   (set-buffer-file-coding-system 'undecided-dos nil))
 
-(defun spacemacs/copy-file ()
-  "Write the file under new name."
-  (interactive)
-  (call-interactively 'write-file))
+(defun spacemacs/copy-file (arg)
+  "Write the file under new name and select action on new file.
+This command prompts for a new filename then prompts for
+selecting an action to perform on the new file.
+
+If prefixed with the universal ARG \\[universal-argument] the
```

Comment:
Can you remind me why we need this prefix argument?
```elisp
(read-file-name "Save file as: " (if arg buffer-file-name default-directory))
```

Can we just write
```elisp
(read-file-name "Save file as: " (or buffer-file-name default-directory))
```

Reply:
This argument is to include the filename in the prompt for new filename like in the [https://github.com/syl20bnr/spacemacs/pull/8974#issuecomment-755299650](original proposal) (but then inverted). The `or` wouldn't work here, at least not for how I meant it. On the other hand `(buffer-name)` would be even better than buffer-file-name.